components/dashboard/src/components/PrebuildLogs.tsx

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import EventEmitter from "events";
+import React, { Suspense, useEffect, useState } from "react";
+import { Workspace, WorkspaceInstance, DisposableCollection, WorkspaceImageBuild, GitpodServer, HEADLESS_LOG_STREAM_STATUS_CODE_REGEX } from "@gitpod/gitpod-protocol";
+import { getGitpodService } from "../service/service";
+
+const WorkspaceLogs = React.lazy(() => import('./WorkspaceLogs'));
+
+export default function PrebuildLogs(props: { workspaceId?: string }) {
+  const [ workspace, setWorkspace ] = useState<Workspace | undefined>();
+  const [ workspaceInstance, setWorkspaceInstance ] = useState<WorkspaceInstance | undefined>();
+  const [ error, setError ] = useState<Error | undefined>();
+  const logsEmitter = new EventEmitter();
+  const service = getGitpodService();
+
+  useEffect(() => {
+    const disposables = new DisposableCollection();
+    (async () => {
+      if (!props.workspaceId) {
+        return;
+      }
+      try {
+        const info = await service.server.getWorkspace(props.workspaceId);
+        if (info.latestInstance) {
+          setWorkspace(info.workspace);
+          setWorkspaceInstance(info.latestInstance);
+        }
+        disposables.push(service.registerClient({
+          onInstanceUpdate: setWorkspaceInstance,
+          onWorkspaceImageBuildLogs: (info: WorkspaceImageBuild.StateInfo, content?: WorkspaceImageBuild.LogContent) => {
+            if (!content) {
+              return;
+            }
+            logsEmitter.emit('logs', content.text);
+          },
+        }));
+        if (info.latestInstance) {
+          disposables.push(watchHeadlessLogs(service.server, info.latestInstance.id, chunk => {
+            logsEmitter.emit('logs', chunk);
+          }, async () => workspaceInstance?.status.phase === 'stopped'));
```

Comment:
@geropl Please double-check here: Is this really what the `checkIfDone` function is supposed to do?

(I'm not exactly sure yet how `watchHeadlessLogs` works exactly.)

Reply:
@jankeromnes Yes, if `workspaceInstance` is updated properly (I assume it is).

Ideally we'd have a API for "getPrebuildState" so we don't have to interpret this from the underlying instance, but the result should be the same.

Reply:
Thanks! (In the future, I think it would be cool if `watchHeadlessLogs` can figure out on its own whenever the job it watches is done or not 😇)